### PR TITLE
PROGRAMMES-5905 redirect /radio/player/<>

### PR DESCRIPTION
**Pull request by tatitati**
First part of this ticket here (it includes same changes in programmes):
https://github.com/bbc/programmes/pull/783

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by votemike** [comment]
LGTM

**Comment by BjornTwachtmann** [comment]
How thoroughly have you checked the new list of endpoints you're adding Francisco?

**Comment by tatitati** [comment]
@BjornTwachtmann I tested them one by one. I deleted around 10 because they didn't work properly:

```diff
    .....
-  'bbc_radio_one_vintage'                 => '/radio1vintage',' // redirect to /programmes/p059cwkn
-  'bbc_7'                                 => '/radio7',  // broken homepage
-  'bbc_radio_cymru_mwy'                   => '/radiocymrumwy', // redirect to /programmes/b07v33kg
-  'bbc_radio_swindon'                     => '/swindon', // redirect to /news/england/wiltshire
-  'bbc_southern_counties_radio'           => '/southerncounties', // redirect to /southerncounties/index.shtml
-  'bbc_music_jazz'                        => '/musicjazz', // redirect to /programmes/p033dmdy
-  'bbc_radio_two_fifties'                 => '/radio250s', // redirect to /programmes/p03lsql7
-  'bbc_radio_scotland_music_extra'        => null, // no network url_key
-  'bbc_radio_two_country'                 => '/radio2country',    redirect to /programmes/p02jxzfq
-  'cbeebies_radio'                        =>'/cbeebiesradio', // redirect to /cbeebies/radio
-  'bbc_radio_two_eurovision'              =>'/radio2eurovision' // redirect to /programmes/p01xjf0z
    ....
```

**Comment by BjornTwachtmann** [comment]
Have you confirmed with product that it's okay to simply stop rendering "listen live" links for those stations? That's not ideal really.

**Comment by tatitati** [comment]
As discussed with Cathal:
Pending to confirm what to do with:
```
    bbc_music_jazz
    bbc_radio_scotland_music_extra
    bbc_radio_two_eurovision
    bbc_southern_counties_radio
```

File: app/config/routing_3rd_party.yml [hunk 1]
```diff
@@ -31,9 +31,9 @@ iplayer_live:
     schemes: [http]
 
 radio_player:
-    path: /radio/player/{sid}
+    path: /{network_urlkey}
```
Comment (felipeparaujo) [inline app/config/routing_3rd_party.yml]: just one small thing: to keep in line with the rest of the routes, camel case this: networkUrlKey. Same this for the test

**Comment by tatitati** [comment]
@BPScott  

Outcome:
```diff
    .....
    // Change for bbc7: 
-  'bbc_7'                                 => '/radio7',
+  'bbc_7'                                 => '/radio4extra',  
    // change for bbc_radio_scotland_music_extra
-  'bbc_radio_scotland_music_extra'        => // no network url_key. Service also finished on Wednesday 28 December 2016, so there is no existing 'listen now/listen live' links for this.
    ....
```

File: app/config/routing_3rd_party.yml [hunk 2]
```diff
@@ -31,9 +31,9 @@ iplayer_live:
     schemes: [http]
 
 radio_player:
-    path: /radio/player/{sid}
+    path: /{networkUrlKey}
```
Comment (votemike) [inline app/config/routing_3rd_party.yml]: What's the difference between this route and `network`? This one looks like a duplication now...

File: src/DsShared/Helpers/LiveBroadcastHelper.php [hunk 3]
```diff
@@ -42,73 +42,73 @@ class LiveBroadcastHelper
...
+        'bbc_1xtra'                             => ['radio_player', ['networkUrlKey' => '1xtra']],
+        'bbc_radio_two'                         => ['radio_player', ['networkUrlKey' => 'radio2']],
+        'bbc_radio_three'                       => ['radio_player', ['networkUrlKey' => 'radio3']],
+        'bbc_radio_fourfm'                      => ['radio_player', ['networkUrlKey' => 'radio4']],
+        'bbc_radio_fourlw'                      => ['radio_player', ['networkUrlKey' => 'radio4']],
+        'bbc_radio_four_extra'                  => ['radio_player', ['networkUrlKey' => 'radio4extra']],
+        'bbc_radio_five_live'                   => ['radio_player', ['networkUrlKey' => '5live']],
+        'bbc_radio_five_live_sports_extra'      => ['radio_player', ['networkUrlKey' => '5livesportsextra']],
+        'bbc_6music'                            => ['radio_player', ['networkUrlKey' => '6music']],
+        'bbc_7'                                 => ['radio_player', ['networkUrlKey' => 'radio4extra']],
```
Comment (votemike) [inline src/DsShared/Helpers/LiveBroadcastHelper.php]: Is this correct? bbc_7 => radio4extra???

Comment (tatitati) [reply src/DsShared/Helpers/LiveBroadcastHelper.php]: yes, discussed with cathal.

File: app/config/routing_3rd_party.yml (hunk 2 above)
Comment (tatitati) [reply app/config/routing_3rd_party.yml]: 😮  nice

**Pull request merged**